### PR TITLE
34 buying appearances

### LINKED ISSUE #34: "Buying" alternate appearances
With the coins the player has gathered, he can buy/unlock new appearances

### PR DESCRIPTION
**Pull request by MKerkhof**
Functionality is now working. You can buy skins and equip them. The unlocked state of the skins is saved to the save file so they stay unlocked for the next time you play.

Will close #34 

<feature

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/gui/Style.java [hunk 1]
```diff
@@ -147,6 +150,17 @@ public static void loadSkins() {
         plank = new PlankSkin();
     }
 
+
+    /**
+     * Method that creates all the skins.
+     * @return ObservableList<Skin> Returns list with skins.
+     */
+    public static ObservableList<Skin> loadSkinsToList() {
+        ObservableList<Skin> list = FXCollections.observableArrayList();
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/Style.java]: `list` could be declared final.

<review,logic

File: src/main/java/nl/tudelft/ti2206/group9/gui/scene/ShopScene.java [hunk 2]
```diff
@@ -32,61 +34,102 @@
...
+     * @return VBox VBox containing an skin item.
      */
-    private static void setUpTable() {
-        items.clear();
-        itemTable.getColumns().clear();
-        itemTable.setItems(items);
-        itemTable.setTooltip(new Tooltip("Double click to change appearance"));
-        items.addAll(Style.getAndy(), Style.getBoy(),
-                Style.getCaptain(), Style.getIronMan(),
-                Style.getNoob(), Style.getPlank());
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/scene/ShopScene.java]: *Auto-generated comment*

    
    Useless parentheses.

**Comment by mpsijm** [comment]
Functionality is looking good :)
I have one question/idea about how you set the locked/unlocked state of the skins.
Currently, only the unlocked field can be set via the entity itself (and thus has to be get via Style). Is it possible to use the Singleton pattern here? Using the Singleton pattern, there is no dependency on Style anymore.

<review,logic,interface

File: src/main/java/nl/tudelft/ti2206/group9/gui/scene/ShopScene.java [hunk 3]
```diff
@@ -32,61 +33,102 @@
...
-                if (event.getClickCount() == 2 && !row.isEmpty()) {
-                    State.setSkin(row.getItem());
+    private static VBox createCarousel(final Skin s) {
+        Label price = createLabel("Price", 0, 0);
+        Label name = createLabel("Name", 0, 0);
+        Button buy = createButton("BUY", 0, 0);
+        if (Skin.getUnlocked(s.getSkinName())) {
+            buy.setText("EQUIP");
+        }
+        buy.setOnAction((event -> {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/scene/ShopScene.java]: *Auto-generated comment*

    
    Useless parentheses.

<review,layout

**Comment by mpsijm** [comment]
I reviewed the last two commits you made, and it's looking good :) I didn't test the functionality anymore, I assume you didn't break it :)
Everyone, please review this, this is a really nice feature :D

<review, interface, organization

**Comment by thervh70** [comment]
I'm missing some kind of warning, in case you dont have enough coins to buy a certain skin. This makes the process slightly easier to understand for the user.
Also i would like to see a different ordering of the skins, ranging from cheap to expensive.
Furthermore I will fix the transparency of Andy and Bman for you.

This is a really good looking feature. I really want this to be merged!

< review, user-interface

**Comment by mpsijm** [comment]
I agree with Robin's points. I would suggest making the button transparent when you cannot buy it.

<review, user-interface

**Comment by mpsijm** [comment]
Thanks for the implementing of the button transparency! Now merge master, fix the EndToEndTest and merge away :D 

<review,testing,feature

**Comment by thervh70** [comment]
Yes this looks good! I`ll merge this!

< review

**Pull request merged**